applications/DEMApplication/python_scripts/DEM_properties_measure_utility.py

```diff
@@ -0,0 +1,196 @@
+            measured_porosity = self.SphericElementGlobalPhysicsCalculator.CalculatePorosityWithinSphere(self.spheres_model_part, radius, [center_x, center_y, center_z])
+            return measured_porosity
+        
+        if type == "averaged_coordination_number":
+            if self.DEM_parameters["ContactMeshOption"].GetBool():
+                averaged_coordination_number = self.ContactElementGlobalPhysicsCalculator.CalculateAveragedCoordinationNumberWithinSphere(self.spheres_model_part, self.contact_model_part, radius, [center_x, center_y, center_z])
+                return averaged_coordination_number
+            else:
+                raise Exception('The \"ContactMeshOption\" in the [ProjectParametersDEM.json] should be [True].')
+        
+        if type == "fabric_tensor":
+            if self.DEM_parameters["ContactMeshOption"].GetBool():
+                measured_fabric_tensor = self.ContactElementGlobalPhysicsCalculator.CalculateFabricTensorWithinSphere(self.contact_model_part, radius, [center_x, center_y, center_z])
+                measured_fabric_tensor = np.array(measured_fabric_tensor)
+                deviatoric_tensor = 4 * (measured_fabric_tensor - 1/3 * np.eye(3)) 
+                second_invariant_of_deviatoric_tensor = (0.5 * np.sum(deviatoric_tensor * deviatoric_tensor))**0.5
+                eigenvalues, eigenvectors = np.linalg.eig(measured_fabric_tensor)
+                return eigenvalues, second_invariant_of_deviatoric_tensor, measured_fabric_tensor
+            else:
+                raise Exception('The \"ContactMeshOption\" in the [ProjectParametersDEM.json] should be [True].')
+            
+        if type == "stress_tensor":
+            if self.DEM_parameters["PostStressStrainOption"].GetBool() and self.DEM_parameters["ContactMeshOption"].GetBool():
+                measure_sphere_volume = 4.0 / 3.0 * math.pi * radius * radius * radius
+                total_stress_tensor = self.ContactElementGlobalPhysicsCalculator.CalculateTotalStressTensorWithinSphere(self.contact_model_part, radius, [center_x, center_y, center_z])
+                averaged_stress_tensor = np.array(total_stress_tensor) / measure_sphere_volume
+                return averaged_stress_tensor
+            else:
+                raise Exception('The \"PostStressStrainOption\" and \"ContactMeshOption\" in the [ProjectParametersDEM.json] should be [True].')
+            
+        if type == "unbalanced_force":
+            if self.DEM_parameters["ContactMeshOption"].GetBool():
+                measured_unbalanced_force = self.ContactElementGlobalPhysicsCalculator.CalculateUnbalancedForceWithinSphere(self.spheres_model_part, self.contact_model_part, radius, [center_x, center_y, center_z])
+                return measured_unbalanced_force
+            else:
+                raise Exception('The \"ContactMeshOption\" in the [ProjectParametersDEM.json] should be [True].')
+        
+        if type == "conductivity_tensor":
```

Comment:
The only improvement I would suggest is to also move this calculation to the C++ class.
In addition, this function is mixing the calculation of the conductivity tensor with the rose diagram; I would make treat them separately, or change the function name to make it clearer.